src/components/SearchFilters/AppliedFilters.tsx

```diff
@@ -0,0 +1,86 @@
+import { useRouter } from "next/router"
+import { TagSet } from "@nypl/design-system-react-components"
+
+import {
+  getQueryWithoutFilters,
+  buildFilters,
+  addLabelPropAndParseFilters,
+} from "../../utils/refineSearchUtils"
+import type { Aggregation, Option } from "../../types/filterTypes"
+
+interface AppliedFiltersPropsType {
+  aggregations: Aggregation[]
+  appliedFilters: Record<string, string[]>
+}
+
+const AppliedFilters = ({
+  appliedFilters,
+  aggregations,
+}: AppliedFiltersPropsType) => {
+  const router = useRouter()
+  const appliedFiltersWithLabels = addLabelPropAndParseFilters(
+    aggregations,
+    appliedFilters
+  )
+  const appliedFilterFields = Object.keys(appliedFiltersWithLabels)
+  const tagSetData = appliedFilterFields
+    .map((field: string) => {
+      const appliedFiltersWithLabelsPerField = appliedFiltersWithLabels[field]
+      return appliedFiltersWithLabelsPerField.map((filter: Option) => {
+        return { id: field + "-" + filter.label, label: filter.label, field }
+      })
+    })
+    .flat()
+
+  const handleRemove = (tagLabel: string) => {
+    if (tagLabel === "clearFilters") {
+      router.push({
+        pathname: "/search",
+        query: getQueryWithoutFilters(router.query),
+      })
+      return
+    }
+    // This click handler only returns the label, but we need the field so
+    // we don't remove the wrong filter
```

Comment:
Yes! Then it would also mirror the checkbox group handler a little more, which returns an array of values. This return an array of labels feels incongruous as well as potentially imprecise

Reply:
@bigfishdesign13 this seems reasonable and easy and while it is, potentially a breaking change, now would be the time to do this.

Reply:
@EdwinGuzman Do you want to squeeze this into DS `v3`? Regarding the update, are suggesting that each internal `Tag` component return an additional data prop value that somehow ties the tag with an associated form field?

Reply:
oh that would be incredible! that would be great functionality for the checkbox group as well.